lib/galaxy/datatypes/binary.py

```diff
@@ -462,6 +467,55 @@ def to_archive(self, trans, dataset, name=""):
         file_paths.append(dataset.metadata.bam_index.file_name)
         return zip(file_paths, rel_paths)
 
+    def get_chunk( self, trans, dataset, offset=0, ck_size=None ):
+        index_file = dataset.metadata.bam_index
+        with pysam.AlignmentFile( dataset.file_name, "rb", index_filename=index_file.file_name ) as bamfile:
+            ck_size = 300  # 300 lines
+            ck_data = ""
+            header_line_count = 0
+            if offset == 0:
+                ck_data = bamfile.text
```

Comment:
what do you think about doing `ck_data = bamfile.text.replace('\t', ' ')` ? That would prevent the header lines to stretch the columns too far (It is a dirty hack ... It would be better to set and render the comment lines in a single cell)